src/platform/plugins/shared/unified_doc_viewer/public/components/doc_viewer_table/table_grid.tsx

```diff
@@ -0,0 +1,322 @@
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type {
+  EuiDataGridCellPopoverElementProps,
+  EuiDataGridProps,
+  EuiDataGridStyle,
+  RenderCellValue,
+  UseEuiTheme,
+} from '@elastic/eui';
+import { EuiCallOut, EuiDataGrid, EuiSpacer, EuiText, euiFontSize } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { useMemoCss } from '@kbn/css-utils/public/use_memo_css';
+import { usePager } from '@kbn/discover-utils';
+import { i18n } from '@kbn/i18n';
+import type { DocViewFilterFn } from '@kbn/unified-doc-viewer/types';
+import React, { useCallback, useMemo } from 'react';
+import { getUnifiedDocViewerServices } from '../../plugin';
+import type { FieldRow } from './field_row';
+import { getPinColumnControl } from './get_pin_control';
+import { TableCell } from './table_cell';
+import {
+  getFieldCellActions,
+  getFieldValueCellActions,
+  getFilterExistsDisabledWarning,
+  getFilterInOutPairDisabledWarning,
+} from './table_cell_actions';
+import type { UseTableFiltersReturn } from './table_filters';
+
+function getGridProps(
+  gridStyle?: EuiDataGridStyle
+): Pick<EuiDataGridProps, 'columnVisibility' | 'rowHeightsOptions' | 'gridStyle'> {
+  return {
+    columnVisibility: {
+      visibleColumns: ['name', 'value'],
+      setVisibleColumns: () => null,
+    },
+    rowHeightsOptions: { defaultHeight: 'auto' },
+    gridStyle: {
+      border: 'horizontal',
+      stripes: true,
+      rowHover: 'highlight',
+      header: 'underline',
+      cellPadding: 'm',
+      fontSize: 's',
+      ...(gridStyle ?? {}),
+    },
+  };
+}
+
+export interface TableGridProps {
+  id: string;
+  containerWidth: number;
+  rows: FieldRow[];
+  isEsqlMode: boolean;
+  filter?: DocViewFilterFn;
+  onAddColumn?: (columnName: string) => void;
+  onRemoveColumn?: (columnName: string) => void;
+  columns?: string[];
+  onFindSearchTermMatch: UseTableFiltersReturn['onFindSearchTermMatch'];
+  searchTerm: string;
+  initialPageSize: number;
+  onChangePageSize?: (newPageSize: number) => void;
+  pinnedFields?: string[];
+  onTogglePinned?: (field: string) => void;
+  hidePinColumn?: boolean;
+  customRenderCellValue?: RenderCellValue;
+  customRenderCellPopover?: React.JSXElementConstructor<EuiDataGridCellPopoverElementProps>;
+  gridStyle?: EuiDataGridStyle;
+  hideDataGridHeader?: boolean;
+}
+
+const MIN_NAME_COLUMN_WIDTH = 150;
+const MAX_NAME_COLUMN_WIDTH = 350;
+export const PAGE_SIZE_OPTIONS = [25, 50, 100, 250, 500];
+
+export const GRID_COLUMN_FIELD_NAME = 'name';
+export const GRID_COLUMN_FIELD_VALUE = 'value';
+
+export function TableGrid({
```

Comment:
Following this [comment](https://github.com/elastic/kibana/pull/231782/files#r2302056266), I think this component should be called only `Table` now. Since this is what it is. It only renders a table.